.github/pull_request_template.md

```diff
@@ -1,23 +1,79 @@
-(e.g. Fixes) followed by the issue number. For more info see:
-https://help.github.com/articles/closing-issues-using-keywords/
+## What does this PR do?
+<!-- _(Mandatory)_
+Replace this comment with a description of what's being changed by this PR. Please explain the WHAT: A clear and concise description of what (patterns used, algorithms implemented, design architecture, message processing, etc.)
+-->
+
+## Checklist
+<!-- _(Mandatory)_
+List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
+-->
+- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
+<!-- 
+Please make sure you have signed our [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). We are not asking you to assign copyright to us, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the CLA once.
+-->
+- [ ] My code follows the [style guidelines of this project](CONTRIBUTING.md#java-language-formatting-guidelines)
+- [ ] I have rebased my changes on top of the latest master branch
+<!--
+Update your local repository with the most recent code from the main repo, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review. As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+-->
+- [ ] I have performed a self-review of my own code
+- [ ] I have made corresponding changes to the documentation
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
+<!--
+Run the test suite to make sure that nothing is broken. See https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing for details.
+-->
+- [ ] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
+- [ ] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
+- [ ] Added an API method or config option? Document in which version this will be introduced.
+- [ ] Added an instrumentation plugin? How did you make sure that old, non-supported versions are not instrumented by accident?
 
-If this pull request is work in progress, create a draft PR instead of prefixing the title with WIP.
+## Author's Checklist
+<!-- _(Recommended)_
+Add a checklist of things that are required to be reviewed in order to have the PR approved
+-->
+
+## How to test this PR locally
```

Comment:
Shouldn't this always be to execute the unit/integration tests?

Reply:
It could be the steps to configure the environment before running the agent, or the application to instrument.

Reply:
If there are any steps besides just executing the tests it would be a bug.

Reply:
Well, the english word `test` is very coupled with the actual automated functions that exercise the production code (the tests 😄). When I say test I mean how to verify that the changes are ok, setting up the local environment for that, and that could be done with a unit/integration test, or with a manual test, because sometimes unit tests are not enough. As an example, here you have an example of this section: https://github.com/mdelapenya/probot-codeowners-labeller/pull/4

Reply:
I'd say in the context of this particular project it would be a rare exception from the rule that everything needs to be able to be verified by unit/integration tests. Thus I'd like to remove this section.

Reply:
Done!